src/ert/run_models/iterated_ensemble_smoother.py

```diff
@@ -130,7 +130,7 @@ def run_experiment(
             posterior = self._storage.create_ensemble(
                 self._experiment_id,
                 name=target_case_format % current_iter,  # noqa
-                ensemble_size=self.ert.getEnsembleSize(),
+                ensemble_size=prior_context.sim_fs.ensemble_size,
```

Comment:
Why do we want to change it? For `prior` we still will have to use `ensemble_size=self.ert.getEnsembleSize()`, or could ensemble size change from prior to posterior if some of the realizations failed?

Reply:
For prior it is fine, as we then start from scratch, however for the posterior we start from the prior, and not the configuration. The size does not change even if realizations fail, but think it is better not to rely on external state.

Reply:
Agere